src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -5936,14 +5908,10 @@ private TypeParameterSyntax ParseTypeParameter()
                 _termState = saveTerm;
             }
 
-            SyntaxToken varianceToken = null;
-            if (this.CurrentToken.Kind == SyntaxKind.InKeyword ||
-                this.CurrentToken.Kind == SyntaxKind.OutKeyword)
-            {
-                varianceToken = CheckFeatureAvailability(this.EatToken(), MessageID.IDS_FeatureTypeVariance);
-            }
-
-            return _syntaxFactory.TypeParameter(attrs, varianceToken, this.ParseIdentifierToken());
+            return _syntaxFactory.TypeParameter(
+                attrs,
+                this.CurrentToken.Kind is SyntaxKind.InKeyword or SyntaxKind.OutKeyword ? EatToken() : null,
```

Comment:
nit: consider naming this argument (`varianceKeyword:`)

Reply:
Personally, it feels self evident to me :-). But that may be because I know the syntax shape really well. Generally though, we don't make these (outside of passing things like 'null' since the shapes are so reasonable.

Reply:
That's okay ("nit" means up to you, not blocking)